fitbenchmarking/results_processing/base_table.py

```diff
@@ -61,296 +66,356 @@ def __init__(self, results, best_results, options, group_dir,
+        str_results = self.get_str_dict(html)
+        row = next(iter(self.sorted_results.values()))
+        minimizers_list = [(r.software, r.minimizer) for r in row]
         columns = pd.MultiIndex.from_tuples(minimizers_list)
-        table.columns = columns
+        table = pd.DataFrame.from_dict(str_results,
+                                       orient='index',
+                                       columns=columns)
         return table
 
-    def to_html(self, table, colour, links, error):
+    def to_html(self):
         """
-        Takes Pandas data frame and converts it into the HTML table output
+        Generate a html version of the table.
 
-        :param table: pandas data frame with from results
-        :type table: Pandas DataFrame
-        :param colour: dictionary containing error codes from the minimizers
-        :type colour: dict
-        :param links: dictionary containing links to the support pages
-        :type links: dict
-        :param error: dictionary containing error codes from the minimizers
-        :type error: dict
-
-        :return: HTLM table output
+        :return: HTML table output
         :rtype: str
         """
+        table = self.create_pandas_data_frame(html=True)
+
+        # Format the table headers
         link_template = '<a href="https://fitbenchmarking.readthedocs.io/'\
                         'en/latest/users/options/minimizer_option.html#'\
                         '{0}-{0}" target="_blank">{0}</a>'
         minimizer_template = '<span title="{0}">{1}</span>'
-        table.apply(lambda x: self.enable_error(x, error, "<sup>{}</sup>"),
-                    axis=1, result_type='expand')
-        table.apply(lambda x: self.enable_link(x, links), axis=1,
-                    result_type='expand')
-
-        minimizers_list = [(link_template.format(s.replace("_", "-")),
-                           minimizer_template.format(
-                               self.options.minimizer_alg_type[m], m))
-                           for s in self.options.software
-                           for m in self.options.minimizers[s]]
+
+        row = next(iter(self.sorted_results.values()))
+        minimizers_list = [
+            (link_template.format(result.software.replace('_', '-')),
+             minimizer_template.format(
+                 self.options.minimizer_alg_type[result.minimizer],
+                 result.minimizer))
+            for result in row]
         columns = pd.MultiIndex.from_tuples(minimizers_list)
         table.columns = columns
 
+        # Format the row labels
         index = []
         for b, i in zip(self.best_results, table.index):
             rel_path = os.path.relpath(path=b.support_page_link,
                                        start=self.group_dir)
             index.append('<a href="{0}">{1}</a>'.format(rel_path, i))
         table.index = index
+        print(table.columns)
+        print(table.index)
```

Comment:
Have these print statements been intentionally left in?

Reply:
Good spot, should be fixed now!